packages/runtime/src/next/client.ts

```diff
@@ -685,12 +685,12 @@ export class Makeswift {
 
     let response
     const responseForRequestedLocale = await this.fetch(
-      `v1/element-trees/${id}?${searchParams.toString()}`,
+      `v1/element-trees/${encodeURIComponent(id)}?${searchParams.toString()}`,
```

Comment:
If the component ID contained slashes, this `fetch` would 404 and we'd always return an empty snapshot.

As an aside, I think we can benefit from using a type-safe library for constructing URL literals, but I couldn't find one with an ergonomic API that was also well-maintained. Out of the two candidates that I found, [`urlcat`](https://github.com/balazsbotond/urlcat) has a bunch of stale deps and hasn't been updated in 2 years, and [`mouri`](https://github.com/Brecert/mouri) is virtually unknown and hasn't been touched for 3 years.

Reply:
Oof. Nice find. I remember being a bit concerned about constructing URLs like this and the possibility of forgetting to `encodeURIComponent`.

I think that is a great idea. An API similar to the motivating example of [Type-Level TypeScript](https://type-level-typescript.com/#chapters) would be great.

```typescript
url('/v1/element-trees/:id', { params: { id }, searchParams })

// or

url('/v1/element-trees/:id?locale=:locale', { id, locale })

// or variants thereof
```